homeassistant/components/fan/xiaomi_miio.py

```diff
@@ -250,30 +487,53 @@ def async_update(self):
 
         try:
             state = yield from self.hass.async_add_job(
-                self._air_purifier.status)
+                self._device.status)
             _LOGGER.debug("Got new state: %s", state)
 
             self._state = state.is_on
-            self._state_attrs = {
+            self._state_attrs.update({
                 ATTR_TEMPERATURE: state.temperature,
                 ATTR_HUMIDITY: state.humidity,
                 ATTR_AIR_QUALITY_INDEX: state.aqi,
                 ATTR_MODE: state.mode.value,
                 ATTR_FILTER_HOURS_USED: state.filter_hours_used,
                 ATTR_FILTER_LIFE: state.filter_life_remaining,
                 ATTR_FAVORITE_LEVEL: state.favorite_level,
-                ATTR_BUZZER: state.buzzer,
                 ATTR_CHILD_LOCK: state.child_lock,
                 ATTR_LED: state.led,
                 ATTR_MOTOR_SPEED: state.motor_speed,
                 ATTR_AVERAGE_AIR_QUALITY_INDEX: state.average_aqi,
                 ATTR_PURIFY_VOLUME: state.purify_volume,
-            }
-
-            if state.led_brightness:
+                ATTR_LEARN_MODE: state.learn_mode,
+                ATTR_SLEEP_TIME: state.sleep_time,
+                ATTR_SLEEP_LEARN_COUNT: state.sleep_mode_learn_count,
+                ATTR_EXTRA_FEATURES: state.extra_features,
+                ATTR_TURBO_MODE_SUPPORTED: state.turbo_mode_supported,
+                ATTR_AUTO_DETECT: state.auto_detect,
+            })
```

Comment:
The same idea that I elaborated above could be use here too, maybe the `available_attributes` could be a map of (attribute_name, attribute_getter)` or similar?

Reply:
Cool! I will care about.